docs/devGuide/projectManagement.md

```diff
@@ -88,6 +88,17 @@ For general best practices, refer to the guide [_Working with PRs_ @SE-EDU](http
    * The end result of this command is version commit with an appropriate tag. We will make use of the generated tag and commit message later.
    </box>
 
+1. **Build the core package** by executing `npm run build:core` in the root directory.
+
+   The script would compile the TypeScript files into JavaScript files (`.js`), sourcemaps (`.js.map`), and type declarations (`.d.ts`), which would be packed when `npm` publishes this package.
```

Comment:
Should all these generated files be committed to the repo? If not, we need some ways to ignore all these files when doing the release commit.

Reply:
The generated files are already in `.gitignore` so hopefully it won't get committed.

Reply:
Oh i was mainly referring to the generated `.js` files from their corresponding `.ts` files. I noticed that these are not ignored in the current `.gitignore`. Another potential issue is that we can't simply ignore all `.js` files in the `core` folder as there will be some files that will not be converted to typescript (as discussed above). Wondering if there is a better way to exclude these files?

Reply:
I see :+1: I just updated the `.gitignore` to compile, the rules are the same as what I have input in `.eslintignore` (ignore all JS files except those we know are still directly written in JS).